### ISSUE TITLE
add support for `gen-interface`

### ISSUE THREAD

**Issue by sumbach** [clj-kondo/clj-kondo#1786]
clj-kondo reports false positive unreesolved-symbol for method names (and possibly argument/return types?) in `(gen-interface ,,,)` forms:

```
clj-kondo --lint - <<'EOF'
(ns a.b.c)
(gen-interface
 :name a.b.c.Foo
 :methods [[bar [String] Object]])
EOF
<stdin>:4:13: error: Unresolved symbol: bar
linting took 11ms, errors: 1, warnings: 0
```

There is no issue with the equivalent `(definterface ,,,)` form:

```
clj-kondo --lint - <<'EOF'
(ns a.b.c)
(definterface Foo
  (^Object bar [^String _]))
EOF
linting took 12ms, errors: 0, warnings: 0
```

It's probably not worth implementing full analysis or linting for `gen-interface`, as folks should probably use `definterface` instead.

Perhaps a reasonable compromise is to skip linting of `(gen-interface ,,,)` forms altogether?

**Comment by sumbach** [comment]
See also https://github.com/clj-kondo/clj-kondo/issues/473#issuecomment-534465132

**Comment by borkdude** [comment]
Agreed, we could probably ignore errors within `gen-interface`. TBH I've never used this construct myself. I would expect not many people cared about correct analysis here.
